Upstream-Status:Inappropriate [embedded specific]

The upstream code assumes that the compiler version used to compiler miniperl/perl-native
is the same as the one being used to build the perl binary. Since most people are not running
systems with gcc 5, it is unlikely that it will work on any supported host. Switch out gccversion
for the version extracted from $CC --version.

Jeremy Puhlman <jpuhlman@mvista.com>

Fix spaces in brackets while running $CC --version
Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 ext/Errno/Errno_pm.PL | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/ext/Errno/Errno_pm.PL b/ext/Errno/Errno_pm.PL
index 1fd29d0..7976ea2 100644
--- a/ext/Errno/Errno_pm.PL
+++ b/ext/Errno/Errno_pm.PL
@@ -224,9 +224,13 @@ sub write_errno_pm {
 
     {	# BeOS (support now removed) did not enter this block
     # invoke CPP and read the output
+        my $compiler = $ENV{'CC'};
+        my $compiler_out = `$compiler --version`;
+        $compiler_out =~ s/\(.*\)/;/;
+        my @compiler_version = split / /,$compiler_out; 
 
 	my $inhibit_linemarkers = '';
-	if ($Config{gccversion} =~ /\A(\d+)\./ and $1 >= 5) {
+	if (@compiler_version[2] =~ /\A(\d+)\./ and $1 >= 5) {
 	    # GCC 5.0 interleaves expanded macros with line numbers breaking
 	    # each line into multiple lines. RT#123784
 	    $inhibit_linemarkers = ' -P';
-- 
1.9.1

